crates/sui-core/src/authority/authority_store_tables.rs

```diff
@@ -147,7 +147,7 @@ impl AuthorityPerpetualTables {
         version: SequenceNumber,
     ) -> Option<Object> {
         let Ok(iter) = self.objects
-            .iter()
+            .range_iter(ObjectKey::min_for_id(&object_id)..=ObjectKey::max_for_id(&object_id))
```

Comment:
Could this just be 
```suggestion        .range_iter(ObjectKey::min_for_id(&object_id)..=ObjectKey(object_id, version))
```
with no `skip_prior_to`?

Reply:
Indeed it can.
I was trying to reduce how much of the non-db code I was changing in this PR.
Other repo-wide improvements will come in a subsequent PR.